test/etools-datepicker_test.html

```diff
@@ -20,10 +21,27 @@
 
     <script>
       suite('etools-datepicker', function() {
+        var basic
+        setup(function() {
+          basic = fixture('basic')
+        });
         test('instantiating the element works', function() {
-          var element = fixture('basic');
-          assert.equal(element.is, 'etools-datepicker');
+          assert.equal(basic.is, 'etools-datepicker');
+        });
+        test('tapping datepicker icon toggles it open', function() {
+          expect(dialog.attributes[4].name).to.equal('aria-hidden')
+          var calendarIcon = Polymer.dom(basic.root).querySelector('iron-icon')
+          MockInteractions.tap(calendarIcon)
+          expect(dialog.attributes[4].name).to.not.equal('aria-hidden')
         });
+        test('tapping a date selects it', function() {
+          var calendarIcon = Polymer.dom(basic.root).querySelector('iron-icon')
+          MockInteractions.tap(calendarIcon)
+          var firstDay = dialog.children.picker.querySelector('.day-item.selection')
```

Comment:
Avoid querying like this `dialog.children.picker.querySelector('.day-item.selection')`. I do not think it's working, at least not with shadowDOM. It should be something like this: `dialog.$$('#picker').$$('#calendar').$$('.day-item.selection')` or `dialog.querySelector('#picker').querySelector('#calendar').querySelector('.day-item.selection')`

Reply:
i think it was working for me, but i see that i can definitely use picker.$$('#calendar').$$('.day-item.selection'), since the <paper-date-picker> element has an id of 'picker'. is that ok?

Reply:
`picker` will not work... It doesn't matter if you have an id `picker` somewhere in element's template... you need to query for that element using this id before you start using it as a variable.

```javascript
// assuming `etoolsDatepicker` is your fixture element
// using `Polymer.dom(etoolsDatepicker.root)` you make sure the search will work well with shadowDom too (shadowDom is not active in this demo element)
let picker = Polymer.dom(etoolsDatepicker.root).querySelector('#picker'); 
// and now `firstDay` will be what you are looking for
let firstDay = picker.$$('#calendar').$$('.day-item.selection');
```